src/lib/logical_query_plan/aggregate_node.cpp

```diff
@@ -77,6 +77,42 @@ std::vector<std::shared_ptr<AbstractExpression>> AggregateNode::output_expressio
   return output_expressions;
 }
 
+std::shared_ptr<LQPUniqueConstraints> AggregateNode::unique_constraints() const {
+  auto unique_constraints = std::make_shared<LQPUniqueConstraints>();
+
+  // (1) Create new unique constraint from the group-by column(s), which form a candidate key for the output relation.
+  const auto group_by_columns_count = aggregate_expressions_begin_idx;
+  ExpressionUnorderedSet group_by_columns(group_by_columns_count);
+  std::copy_n(node_expressions.begin(), group_by_columns_count,
+              std::inserter(group_by_columns, group_by_columns.begin()));
+
+  unique_constraints->emplace_back(group_by_columns);
+
+  /**
+   * (2) Forward unique constraints from child nodes, if all expressions belong to the group-by section.
+   *     Note: The DependentGroupByReductionRule might wrap some expressions with an ANY() aggregate function.
+   *     However, ANY() is a pseudo aggregate function and does not change values (cf. DependentGroupByReductionRule).
+   *     Therefore, ANY()-wrapped columns can be interpreted as group-by columns.
+   */
+
+  // Check each constraint for applicability
+  const auto& input_unique_constraints = left_input()->unique_constraints();
+  for (const auto& input_unique_constraint : *input_unique_constraints) {
+    if (!has_output_expressions(input_unique_constraint.expressions)) continue;
+
+    // Avoid duplicates
+    if (std::any_of(unique_constraints->cbegin(), unique_constraints->cend(),
+                    [&input_unique_constraint](const auto& unique_constraint) {
+                      return input_unique_constraint == unique_constraint;
+                    }))
```

Comment:
If we can trust the input to not produce duplicates, we should only have to check for equality to `unique_constraints`, i.e., the unique constraint created from the GROUP BY columns.

Reply:
I swapped (1) and (2) and moved the duplicate check into "(2) Create new unique constraint".  

Should save some cycles.

Reply:
Ok, I was able to improve it even further: 

* MockNode: a, b, c
* MockNode Unique Constraint: a
* AggregateNode: GroupBy(a, b)

We should not create a unique constraint from (a, b) because we already have one for a. I also added a test for that.